thorlcr/master/thmastermain.cpp

```diff
@@ -358,6 +357,47 @@ class CRegistryServer : public CSimpleInterface
                     EXCLOG(e, "Shutting down slave");
                     e->Release();
                 }
+                if (watchdog)
+                    watchdog->removeSlave(ep);
+            }
+        }
+
+        CTimeMon tm(20000);
+        unsigned numReplied = 0;
+        while (numReplied < slavesRegistered)
+        {
+            unsigned remaining;
+            if (tm.timedout(&remaining))
+            {
+                PROGLOG("Timeout waiting for Shutdown reply from slave(s) (%u replied out of %u total)", numReplied, slavesRegistered);
+                StringBuffer slaveList;
+                for (i=0;i<slavesRegistered;i++)
+                {
+                    if (status->test(i))
+                        slaveList.append(i+1).append(",");
+                }
+                if ((i=slaveList.length()))
```

Comment:
think a tad confusing (I thought was typo')

I think above better if you add comma in if (slaveList.length()) test. 
At expense of 1 extra if in non-critical code, it will look cleaner.